atomicfu/src/commonMain/kotlin/kotlinx/atomicfu/locks/Mutex.kt

```diff
@@ -0,0 +1,28 @@
+package kotlinx.atomicfu.locks
+
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * Multiplatform mutex.
+ * On native based on pthread system calls.
+ * On JVM delegates to ReentrantLock.
```

Comment:
This doesn't seem accurate, though? `ReentrantLock` does get used in `synchronized` and `kotlinx.atomicfu.ReentrantLock`, but not in `Mutex`, right?

If so, there seems to be some inconsistency: on the one hand, `atomicfu.ReentrantLock` delegates to Java's `ReentrantLock`, on the other, `atomicfu.Mutex` doesn't, whereas on Native, `atomicfu.ReentrantLock` and `atomcifu.Mutex` are basically the same thing.
If we believe that our mutex is better, we should use it throughout; if we don't, we should use `ReentrantLock` throughout. Are there measurements showing which one performs better?